MetaMask/metamask-extension
# feat: carousel component

## **Description**

This PR introduces a carousel component for the homepage that displays banners about new changes. It leverages the existing `react-responsive-carousel` library to handle carousel functionality and the `BannerBase` component for rendering banner content and images. The carousel component has been integrated into the `account-overview-layout.tsx` file.

Key features:
- Skeleton is displayed during app startup or if there is no connection.
- Implements banner-specific behaviors based on user interaction (e.g., navigation dots, horizontal drag, close button).
- Supports up to 5 banners with no auto-forwarding or loop functionality.
- Non-dismissible 'Fund your wallet' banner remains in the first position when applicable.

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/3758

## **Manual testing steps**

1. Start the app and navigate to the homepage.
2. Verify that the carousel skeleton appears during app initialization or when there’s no connection.
3. Test navigation between banners using dots, horizontal drag, and clicking edges.
4. Verify that banners behave as expected when interacting with the close button:
   - For last banners, ensure the container is removed.
   - For 'Fund your wallet' banner, ensure it remains in place when others are closed.
5. Confirm URLs open correctly when clicking Bridge, Funds, Sell, and Card banners.
6. Test with more than 5 banners to ensure only the first 5 are displayed.
7. Verify no auto-forwarding or loop behavior.

## **Screenshots/Recordings**

### **Before**



### **After**

<img width="356" alt="Screenshot 2024-12-09 at 22 38 04" src="https://github.com/user-attachments/assets/103aefff-a7e4-4293-a19a-19f8b0830e67">


## **Pre-merge author checklist**

- [ ] I’ve included tests for the carousel component and its behaviors.
- [ ] I’ve applied the appropriate labels as per the [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md).

## **Pre-merge reviewer checklist**

- [ ] I’ve manually tested the PR by pulling the branch, running the app, and verifying functionality.
- [ ] I confirm that this PR meets all acceptance criteria from the ticket.
- [ ] I’ve reviewed and approved the testing evidence provided (screenshots/recordings).
